src/test/datascience/plotViewer/plotViewer.vscode.test.ts

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { assert } from 'chai';
+import * as vscode from 'vscode';
+import { IVSCodeNotebook } from '../../../client/common/application/types';
+import { traceInfo } from '../../../client/common/logger';
+import { IDisposable } from '../../../client/common/types';
+import { IExtensionTestApi, waitForCondition } from '../../common';
+import { closeActiveWindows, initialize } from '../../initialize';
+import {
+    closeNotebooksAndCleanUpAfterTests,
+    insertCodeCell,
+    startJupyterServer,
+    waitForExecutionCompletedSuccessfully,
+    createEmptyPythonNotebook,
+    runAllCellsInActiveNotebook,
+    canRunNotebookTests
+} from '../notebook/helper';
+
+suite('VSCode Notebook PlotViewer integration - VSCode Notebook', function () {
+    let api: IExtensionTestApi;
+    let vscodeNotebook: IVSCodeNotebook;
+    const disposables: IDisposable[] = [];
+    // On conda these take longer for some reason.
+    this.timeout(120_000);
+
+    suiteSetup(async function () {
+        api = await initialize();
+        if (!(await canRunNotebookTests())) {
+            return this.skip();
+        }
+        await startJupyterServer();
+        vscodeNotebook = api.serviceContainer.get<IVSCodeNotebook>(IVSCodeNotebook);
+    });
+
+    setup(async function () {
+        traceInfo(`Start Test ${this.currentTest?.title}`);
+        traceInfo(`Start Test Completed ${this.currentTest?.title}`);
+    });
+
+    teardown(async function () {
+        traceInfo(`End Test ${this.currentTest?.title}`);
+        await closeNotebooksAndCleanUpAfterTests(disposables);
+        traceInfo(`End Test Completed ${this.currentTest?.title}`);
+    });
+    suiteTeardown(() => closeNotebooksAndCleanUpAfterTests(disposables));
+
+    test('Verify plot viewer is active for PNG plots', async function () {
+        await startJupyterServer();
+        await closeActiveWindows();
+        await createEmptyPythonNotebook(disposables);
+        assert.isOk(vscodeNotebook.activeNotebookEditor, 'No active notebook');
```

Comment:
I think the first 4 lines can be moved into `setup`.
Then we don't need to repeat this in every test

Reply:
@DonJayamanne I actually explicitly moved it out since I needed to change the VS Code settings in the second test before opening and running the file.

Reply:
Guess I could put in a comment to that effect.